src/utils/readOnlyUtils.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/*
+    This functionality is more or less "cherry-picked" from package "vscode-azureextensionui".
+    Currently we can not use it from there due to the fact that we depend on an old version.
+    Once we move to a more recent version we should eventually remove this file and rewire the
+    call sites to the "genuine" implementation.
+*/
+
+import { CancellationToken, Event, EventEmitter, TextDocumentContentProvider, Uri, window, workspace } from "vscode";
```

Comment:
(nit) Use VSCode's "Organize imports" command to sort lines by import source and imports in each line.